src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1411,11 +1411,11 @@ impl EventProcessor {
         // Set the timestamp.
         wt.xconn.set_timestamp(xev.time as xproto::Timestamp);
 
-        let did = mkdid(xev.deviceid as xinput::DeviceId);
+        let device_id = mkdid(xev.sourceid as xinput::DeviceId);
 
         let mask =
             unsafe { slice::from_raw_parts(xev.valuators.mask, xev.valuators.mask_len as usize) };
-        let mut value = xev.raw_values;
+        let mut value = xev.valuators.values;
```

Comment:
What is the difference between `raw_values` and `valuators.values`? [This document](https://www.x.org/releases/X11R7.7/doc/inputproto/XI2proto.txt) says this:

> A RawEvent provides the information provided by the driver to the
client. RawEvent provides both the raw data as supplied by the driver and
transformed data as used in the server. Transformations include, but are
not limited to, axis clipping and acceleration.

I agree with the idea that we should be using these values instead of the raw driver values. However I'm not entirely sure what "axis clipping" is. Google doesn't have many results and all I see is [this release](https://answers.launchpad.net/ubuntu/+source/xorg-server/2:1.3.0.0.dfsg-12ubuntu3):

> 231_In_dmxBackendMouGetInfo_initialize_the_info-> minval_maxval_arrays_to_the_size_of_the_backend_display.patch:
      Fixes potential issue in X input where axis clipping code in
      GetPointerEvents() constrains the pointer's coordinate range to a
      max of 0, causing the mouse to not move.

Eventually I found [this commit](https://gitlab.freedesktop.org/xorg/xserver/-/commit/0aaf28e5633a59563b89a2e42d19fabc84adc3ed), which appears to be somehow clipping the mouse valuators to the monitor's size. So I guess that's what is happening here; the raw values don't keep the monitor border in mind. Hence why it gives us garbage every now and then.

Reply:
I don't know if its actually the raw input, as my pointing device reports different values when looking at the raw hid reports. raw_values just seem to be either -1.0, 0.0 and 1.0, it doesn't make a difference how fast its moving, which is not usable.

But its also not ideal if it is affected by acceleration, as that's the main reason for using raw input.